### PR TITLE
Fix auth adapter loading without App.namespace

### LINKED ISSUE #16258: [cakephp/http] Auth adapters can't be used
### Description

Using `cakephp/http` standalone, with any auth adapter, currently doesn't work without setting `App.namespace` during bootstrap.

```
PHP Fatal error:  Uncaught TypeError: rtrim() expects parameter 1 to be string, null given in vendor/cakephp/core/App.php:63
```

We could either add a fallback in https://github.com/cakephp/cakephp/blob/4.x/src/Core/App.php#L62 to `App` or check for a null `$base` value before applying `rtrim`.

### CakePHP Version

4.3.4

### PHP Version

7.4

### PR DESCRIPTION
**Pull request by cleptric**
Fixes #16258

### CODE REVIEW

File: tests/TestCase/Http/ClientTest.php [hunk 1]
```diff
@@ -1157,4 +1158,41 @@ public function testAddMockResponseCustomInvalidDecision(): void
 
         $client->post('http://example.com/path');
     }
+
+    /**
+     * Check that auth adapters can be used without a set App.namespace
+     * config value.
+     */
+    public function testNoNameSpace()
```
Comment (ADmad) [inline tests/TestCase/Http/ClientTest.php]: Could have saved yourself quite some effort by just adding a test for `App::className()` :)

Comment (ADmad) [reply tests/TestCase/Http/ClientTest.php]: This isn't exactly a unit test for App::className() either.

Comment (markstory) [inline tests/TestCase/Http/ClientTest.php]:
```suggestion
    public function testAuthNoNameSpace()
```

Comment (ADmad) [reply tests/TestCase/Http/ClientTest.php]: > This isn't exactly a unit test for App::className() either.

I have added one. Given that the root cause of the problem is now tested for I don't think this test is necessary, it just adds bloat.

Comment (cleptric) [reply tests/TestCase/Http/ClientTest.php]: I wanted to replicate my exact issue, which wasn't a lot of effort, as this was a copy of a different test.
Anyway, I agree that testing the actual method makes more sense, so I removed my test again.

**Pull request merged**